_assets/js/main.js

```diff
@@ -931,59 +1032,166 @@ jQuery(document).ready(function($) {
-      //var result = index.search(query); --> this will be implemented when we integrate lunr js
-      var num = 0; //result.length && query != '' ? result.length : 0;
-      var query_output = num + ' results for "' + query + '"';
+      var result = idx.search(query);
+      var num = result.length && query != '' ? result.length : 0;
+      var query_output =
+        num + ' ' + $container.data('message') + ' "' + query + '"';
       var $title = $('<h2 class="no-underline"></h2>');
 
-      // Show results
-      $container.empty();
-
-      // Add status
-      $title.text(query_output);
-      $container.prepend($title);
-
-      if (num != 0) {
-        var $list = $('<ol></ol>');
-
-        // Loop through, match, and add results --> this will be implemented when we integrate lunr js
-        // for (var item in result) {
-        //   var ref = result[item].ref;
-        //   var topic = store[ref].topic
-        //     ? '<p class="post-meta"><small>' + store[ref].topic + '</small></p>'
-        //     : '';
-        //   var searchitem =
-        //     '<li class="result"><a href="' +
-        //     store[ref].link +
-        //     '">' +
-        //     topic +
-        //     '<h3>' +
-        //     store[ref].title +
-        //     '</h3><p>' +
-        //     store[ref].excerpt +
-        //     '</p><p><small>' +
-        //     store[ref].link +
-        //     '</small></p></a></li>';
-
-        //   $list.append(searchitem);
-        // }
-
-        $default.hide(0);
-        $container.append($list);
-      } else {
-        $default.show(0);
-      }
+      data.then(function(loaded_data) {
+        // Show results
+        $container.empty();
+
+        // Add status
+        $title.text(query_output);
+        $container.prepend($title);
+
+        if (num != 0) {
+          var $list = $('<ol></ol>');
+
+          // Loop through, match, and add results
+          for (var item in result) {
+            var ref = result[item].ref;
+            var item = loaded_data.entries[ref];
+            var topic =
+              item.meta && item.meta.topic
+                ? '<p class="post-meta"><small>' +
+                  item.meta.topic +
+                  '</small></p>'
+                : '';
+            var excerpt =
+              $.trim(item.body)
+                .substring(0, 300)
+                .split(' ')
+                .slice(0, -1)
+                .join(' ') + '...';
+            var searchitem =
+              '<li class="result"><a href="' +
+              item.url +
+              '">' +
+              topic +
+              '<h3>' +
+              item.title +
+              '</h3><p>' +
+              excerpt +
+              '</p><p><small>' +
+              window.location.origin +
+              item.url +
+              '</small></p></a></li>';
+
+            $list.append(searchitem);
+          }
+
+          $default.hide(0);
+          $container.append($list);
```

Comment:
Any markup here in the data could mess up the layout, it would probably be a good idea to escape the parts used to build up the HTML.

Reply:
It looks like this content is already cleaned when it is compiled into the pages.json file. Do you think it should still be done here as well? From filters.rb in the Jekyll pages api plugin: 
```
def text_only(str)
      # apply each filter in order
      [
        :strip_html,
        :condense,
        :decode_html,
        :strip_html,
        :condense
      ].reduce(str) do |result, filter|
        self.send(filter, result)
      end
    end
```

Reply:
If that's the case this should be fine.